src/Storages/StorageURL.cpp

```diff
@@ -199,6 +202,31 @@ namespace
             };
         }
 
+        // decode a url-encoded string
+        String urlDecode(String str){
```

Comment:
There is already a similar implementation [decodeURL](https://github.com/ClickHouse/ClickHouse/blob/master/src/Functions/URL/decodeURLComponent.cpp#L15) for the DecodeURLComponent ClickHouse function. I wonder the same logic can be reused.

Reply:
Absolutely agree! I'll remove my function.

Reply:
Oh, I was suggesting that you could extract the same implementation (which is already available) and add to `StorageURL.cpp`.

Reply:
Hi @bharatnc , I replaced url decoding function as your suggestion.  Please check!

Reply:
Our test environment doesn't support docker any longer? I see a lot of fails caused by missing docker command